tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -150,60 +159,113 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
-                    assertVisibleTips("There shouldn't be any tooltips visible when the currentPage is focused", []);
-                } else {
-                    assertVisibleTips("Only the tooltip for page link " + (idx + 1) + ", in the " + location + " page bar is visible", [linkId]);
-                }
-                sequence.push({
-                    element: link,
-                    jQueryTrigger: "blur"
-                });
-                assertVisibleTips("There shouldn't be any tooltips visible when none of the pageLinks are focused", []);
-            };
-        };
+    /**
+     * Change the page size
+     */
+    fluid.tests.setPageSize = function (pager, sequence, pageSize) {
+        sequence.push({
+            func: function () {
+                pager.locate("pageSize").val(pageSize);
+            }
+        });
+        sequence.push({
+            element: "#rendered-ioc .flc-pager-page-size",
+            jQueryTrigger: "change"
+        });
+    };
 
-        pageLinksTop.each(tooltipTest("top"));
-        pageLinksBottom.each(tooltipTest("bottom"));
-        return {
-            name: "Pager tooltip tests",
-            tests: {
-                name: "Tooltip visibility",
-                sequence: sequence
+    /**
+     * verify whether the displaying links and their tooltips
+     */
+    fluid.tests.verifyToolTips = function (pager, sequence, pageSize, expectedTooltips) {
+        var expectedNumberOfLinks = Math.ceil(pager.dataModel.length / pageSize);
+        sequence.push({
+            func: function () {
+                pager.locate("pageSize").val(pageSize);
+                jqUnit.assertEquals("When the page size is " + pager.model.pageSize + ", Top pageLinks",
+                    expectedNumberOfLinks,
+                    $("#rendered-ioc .flc-pager-top .flc-pager-links .flc-pager-pageLink a").length);
+                jqUnit.assertEquals("When the page size is " + pager.model.pageSize + ", Bottom pageLinks",
+                    expectedNumberOfLinks,
+                    $("#rendered-ioc .flc-pager-bottom .flc-pager-links .flc-pager-pageLink a").length);
             }
-        };
+        });
+
+        for (var x = 0; x < expectedNumberOfLinks; x++) {
+            fluid.tests.verifyToolTip(pager, sequence, "top", x, expectedTooltips[x]);
+            fluid.tests.verifyToolTip(pager, sequence, "bottom", x, expectedTooltips[x]);
+        }
+    };
+
+    /**
+     * Verify whether the tooltip of the given index is matched with the provided tooltip
+     */
+    fluid.tests.verifyToolTip = function (pager, sequence, location, index, expectedTooltip) {
+        var link =  "#rendered-ioc .flc-pager-" + location + " .flc-pager-links .flc-pager-pageLink a:eq(" + index + ")";
+        var linkNumber = index + 1;
+        var linkId = "page-link:link" + linkNumber + (location === "bottom" ? "-1" : "");
```

Comment:
Similar to above, you should avoid constructing these jQuery selectors. Also I think you should just pass in the pageBar component you are interested in, instead of passing in the location.